test/test_bulk.py

```diff
@@ -347,7 +347,7 @@ def test_bulk_write_no_results(self):
 
     def test_bulk_write_invalid_arguments(self):
         # The requests argument must be a list.
-        generator = (InsertOne({}) for _ in range(10))
+        generator = (InsertOne[dict]({}) for _ in range(10))
```

Comment:
Why are these changes required? This seems annoyingly verbose for users.

Reply:
For the same reason we need a type annotation on MongoClient